go/plugins/dotprompt/genkit.go

```diff
@@ -174,9 +171,16 @@ func (p *Prompt) Register() error {
 // the prompt.
 //
 // This implements the [ai.Prompt] interface.
-func (p *Prompt) Generate(ctx context.Context, pr *PromptRequest, cb func(context.Context, *ai.ModelResponseChunk) error) (*ai.ModelResponse, error) {
+func (p *Prompt) Generate(ctx context.Context, pr *PromptRequest, opts ...PromptOption) (*ai.ModelResponse, error) {
```

Comment:
We want to remove the `PromptRequest` arg and replace it with options only. Also, we want to have a different set of options for `Generate()` since not everything that makes sense at definition makes sense at generation. For example, the input/output schema is relative to the prompt and what it expects so it makes sense for it to be set at definition only.

At generation, we only want what's in `PromptRequest' + streaming. Streaming option should be removed from `PromptOption`.